odoo/addons/base/models/res_bank.py

```diff
@@ -61,7 +61,7 @@ class ResPartnerBank(models.Model):
     _name = 'res.partner.bank'
     _rec_name = 'acc_number'
     _description = 'Bank Accounts'
-    _order = 'sequence, id'
+    _order = 'sequence, id desc'
```

Comment:
Good point, my approach is too heavy-handed (even more so for migrated databases). I'm instead explicitly setting the sequence now:

```python
if partner.bank_ids:
    values['sequence'] = min(partner.bank_ids.mapped("sequence")) - 1
```

It may result in negative sequences but I think those work fine and it avoids having to re-sequence every bank account.